### ISSUE TITLE
SSE not working on Safari iOS 17.4

### ISSUE THREAD

**Issue by soyangel** [bigskysoftware/htmx#2388]
Hi,

It seems like HTMX SSE is not working on Safari in iOS 17.4 (tested with iPhone SE 3rd gen). It was working OK on iOS 17.3.1

Don't know if I'm doing anything wrong or there is something fishy with iOS 17.4

I have uploaded a minimal reproducible example on https://chat.cercaapp.com/static/bug.html with the following code:

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>SSE Bug</title>
    <script src="https://unpkg.com/htmx.org@1.9.10"
        integrity="sha384-D1Kt99CQMDuVetoL1lrYwg5t+9QdHe7NLX/SoJYkXDFfX37iInKRy5xLSi8nO7UC"
        crossorigin="anonymous"></script>
    <script src="https://unpkg.com/htmx.org/dist/ext/sse.js"></script>
</head>

<body>
    <div hx-ext="sse" sse-connect="/sse" sse-swap="message" hx-swap="beforeend">
    </div>
</body>

</html>
```

Also I have an example that works on iOS 17.4 using EventSource on https://chat.cercaapp.com/static/nobug.html with the code:
``` html

<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>SSE NO Bug</title>
</head>

<body>
    <div id="messages"></div>
    <script>
        const events = new EventSource("/sse");
        events.onmessage = (msg) => {
            console.log(msg);
            let div = document.getElementById("messages");
            let text = div.innerHTML + msg.data;
            div.innerHTML = text;
        };
    </script>

</body>

</html>
```

**Comment by artemgordinskiy** [comment]
This might be related to #2389

**Comment by artemgordinskiy** [comment]
I think the fix in #2391 could solve this issue as well

**Comment by Renerick** [comment]
i'll update the pr to fix sse as well
